my-webxr-app/tests/DataPoint.test.tsx

```diff
@@ -41,16 +38,95 @@ describe("DataPoint Creation and Interaction", () => {
     const renderer = await ReactThreeTestRenderer.create(
       <PointSelectionProvider>
         <XR>
-          <DataPoint id={0} outlineScale={2} />
+          <DataPoint id={0} marker={"circle"} color={"gray"} column1={"John Doe"} column2={"cmpt 145"} column3={97} outlineScale={2} />
         </XR>
       </PointSelectionProvider>,
     );
 
-    // Same tests and check if the outline mesh scale is accurate.
-    expect(renderer.scene.children.length).toEqual(2); // + native camera component = 2
-    expect(renderer.scene.children[1].children.length).toEqual(2);
+    // Check if the outline mesh scale is accurate.
     expect(renderer.scene.children[1].children[1].instance.scale).toEqual(
       new Vector3(2, 2, 2),
     );
   });
 });
+
+
+describe("DataPoint UI Interaction", () => {
+  test("create a basic DataPoint and check id", async () => {
+    const renderer = await ReactThreeTestRenderer.create(
+        <PointSelectionProvider>
+          <XR>
+            <DataPoint id={0} marker={"circle"} color={"gray"} column1={"John Doe"} column2={"cmpt 145"} column3={97} />
+          </XR>
+        </PointSelectionProvider>,
+    );
+
+    // Check if the point id value is accurate.
+    expect(renderer.scene.children[1].children[0].instance.userData.id).toBe(0)
```

Comment:
gurll! You got this things figured out 🥳 
It'll be nice to have some docs/comments on what is `renderer.scene.children[1]` or `renderer.scene.children[1].children[0]` so it's easier for other devs to read the tests!
Also for this "DataPoint UI Interaction" set of tests, everything in here can be combined into 1 test which has multiple expect() calls

Reply:
Yes, thanks to Zander's help 🙌

For sure, I will make those changes!